releasey/libs/_version.sh

```diff
@@ -0,0 +1,176 @@
+  # Otherwise, it returns 1.
+  local version="$1"
+
+  if [[ ! ${version} =~ ${BRANCH_VERSION_REGEX} ]]; then
+    return 1
+  fi
+
+  major="${BASH_REMATCH[1]}"
+  minor="${BASH_REMATCH[2]}"
+  # patch is not set from branch name since it uses "x" placeholder
+
+  return 0
+}
+
+function validate_and_extract_git_tag_version {
+  # This function validates the format of a git tag version and extracts its components (major.minor.patch and rc number).
+  # It is similar to validate_and_extract_rc_version, but for git tag format.
+  # It returns 0 if the version is valid and sets the global variables major, minor, patch, and rc_number.
+  # It also sets the global variable version_without_rc to the "major.minor.patch-incubating" format without the rc number.
+  # Otherwise, it returns 1.
+  local version="$1"
+
+  if [[ ! ${version} =~ ${VERSION_REGEX_GIT_TAG} ]]; then
+    return 1
+  fi
+
+  major="${BASH_REMATCH[1]}"
+  minor="${BASH_REMATCH[2]}"
+  patch="${BASH_REMATCH[3]}"
+  rc_number="${BASH_REMATCH[4]}"
+  version_without_rc="${major}.${minor}.${patch}-incubating"
+
+  return 0
+}
+
+function validate_and_extract_polaris_version {
+  # This function validates the format of a Polaris version and extracts its components (major.minor.patch).
+  # It accepts the full version format (e.g., "1.0.0-incubating") and sets the global variables major, minor, patch.
+  # It also sets the global variable version_without_rc to the "major.minor.patch-incubating" format.
+  # Returns 0 if the version is valid, 1 otherwise.
+  local version="$1"
+
+  if [[ ! ${version} =~ ${VERSION_REGEX} ]]; then
+    return 1
+  fi
+
+  major="${BASH_REMATCH[1]}"
+  minor="${BASH_REMATCH[2]}"
+  patch="${BASH_REMATCH[3]}"
+  version_without_rc="${major}.${minor}.${patch}-incubating"
+
+  return 0
+}
+
+function update_version {
+  local version="$1"
+  local current_version=$(cat "$VERSION_FILE")
+  update_version_txt "${version}"
+  update_helm_version "${current_version}" "${version}"
+}
+
+function update_version_txt {
+  local version="$1"
+  # This function is only there for dry-run support.  Because of the
+  # redirection, we cannot use exec_process with the exact command that will be
+  # executed.
+  if [[ ${DRY_RUN:-1} -ne 1 ]]; then
+    exec_process echo ${version} >$VERSION_FILE
+  else
+    exec_process "echo ${version} > $VERSION_FILE"
+  fi
+}
+
+function update_helm_version {
+  local old_version="$1"
+  local new_version="$2"
+  exec_process sed -i~ "s/${old_version}/${new_version}/g" "$HELM_CHART_YAML_FILE"
+  exec_process sed -i~ "s/${old_version}/${new_version}/g" "$HELM_README_FILE"
+  # The readme file may contain version with double dash for shields.io badges
+  # We need a second `sed` command to ensure that the version replacement preserves this double-dash syntax.
+  local current_version_with_dash=$(echo "$old_version" | sed 's/-/--/g')
+  local version_with_dash=$(echo "$version" | sed 's/-/--/g')
```

Comment:
Nit: simplify ((SC2001)[https://github.com/koalaman/shellcheck/wiki/SC2001])
```suggestion
  local current_version_with_dash
  local version_with_dash
  current_version_with_dash="$(echo "${old_version//-/--/g}")"
  version_with_dash="$(echo "$version//-/--/g}")"
```